logger_test.go

```diff
@@ -401,6 +410,15 @@ func TestLoggerHooks(t *testing.T) {
 	assert.Equal(t, int64(2), seen.Load(), "Hook saw an unexpected number of logs.")
 }
 
+func TestLoggerHooksWithFields(t *testing.T) {
+	hook, seen := makeCountingHookWithFields()
```

Comment:
The counting hooks doesn't validate the fields, which would be useful since that is the main difference between `TestLoggerHooks` and this test. Given that `countingHookWithFields` is only used once, it may be better to inline that function here, and validate the entry + fields